src/main/webapp/app/shared/user-settings/ide-preferences/ide-settings.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Ide, IdeMappingDTO } from 'app/shared/user-settings/ide-preferences/ide.model';
+import { ProgrammingLanguage } from 'app/entities/programming-exercise.model';
+
+@Injectable({ providedIn: 'root' })
+export class IdeSettingsService {
+    public ideSettingsUrl = 'api/ide-settings';
+    error?: string;
+
+    programmingLanguageToIde: Map<ProgrammingLanguage, Ide>;
```

Comment:
I'd prefer the service to be stateless & for the state to be kept in the components that use this service.

AFAIK this service will be instantiated once -> if the user logs out, and a different user logs in, the state may not be refreshed and the user might see the wrong preferences.

Reply:
resolved